pallets/runtime/tests/src/settlement_test.rs

```diff
@@ -3066,3 +3083,371 @@ fn multiple_custodian_settlement() {
             );
         });
 }
+
+// ----------------------------------------- Confidential transfer tests -----------------------------------
+
+pub fn gen_account(
+    tx_id: u32,
+    seed: [u8; 32],
+    token_name: &[u8],
+    valid_asset_ids: Vec<AssetId>,
+) -> (SecAccount, PubAccountTx) {
+    // These are the encryptions keys used by MERCAT and are different from the signing keys
+    // that Polymesh uses.
+    let mut rng = StdRng::from_seed(seed);
+    let elg_secret = ElgamalSecretKey::new(Scalar::random(&mut rng));
+    let elg_pub = elg_secret.get_public_key();
+    let enc_keys = EncryptionKeys {
+        pblc: elg_pub.into(),
+        scrt: elg_secret.into(),
+    };
+
+    let asset_id = AssetId {
+        id: *Ticker::try_from(token_name).unwrap().as_bytes(),
+    };
+    let asset_id_witness = CommitmentWitness::from((asset_id.clone().into(), &mut rng));
+    let secret_account = SecAccount {
+        enc_keys,
+        asset_id_witness,
+    };
+    let valid_asset_ids = convert_asset_ids(valid_asset_ids);
+    let mercat_account_tx = AccountCreator {}
+        .create(tx_id, &secret_account, &valid_asset_ids, &mut rng)
+        .unwrap();
+
+    (secret_account, mercat_account_tx)
+}
+
+pub fn init_account(
+    tx_id: u32,
+    seed: [u8; 32],
+    token_name: &[u8],
+    owner: Public,
+) -> (SecAccount, EncryptedAssetIdWrapper) {
+    let valid_asset_ids: Vec<AssetId> = ConfidentialAsset::confidential_tickers();
+    let (scrt_account, mercat_account_tx) = gen_account(tx_id, seed, token_name, valid_asset_ids);
+
+    ConfidentialAsset::validate_mercat_account(Origin::signed(owner), mercat_account_tx.clone())
```

Comment:
I assume you are calling a dispatchable here, If yes then it should be an `assert_ok` statement to verify whether it successfully executed or not. unwrap() panics if Err occurs.